File: system-test/dotnet/Driver.Adapter/External/Clock/Client/ClockRealClient.cs
using Common;
using Driver.Adapter.External.Clock.Client.Dtos;
using Driver.Adapter.External.Clock.Client.Dtos.Error;

namespace Driver.Adapter.External.Clock.Client;

public class ClockRealClient
{
    public static Task<Result<VoidValue, ExtClockErrorResponse>> CheckHealthAsync()
    {
        var _ = Now;
        return Task.FromResult(Result.Success<ExtClockErrorResponse>());
    }

    public Task<Result<ExtGetTimeResponse, ExtClockErrorResponse>> GetTimeAsync()
        => Task.FromResult(Result<ExtGetTimeResponse, ExtClockErrorResponse>.Success(
            new ExtGetTimeResponse { Time = Now }));

    private static DateTimeOffset Now => DateTimeOffset.UtcNow;
}

File: system-test/java/src/main/java/com/mycompany/myshop/testkit/driver/adapter/external/clock/client/ClockRealClient.java
package com.mycompany.myshop.testkit.driver.adapter.external.clock.client;

import com.mycompany.myshop.testkit.common.Result;
import com.mycompany.myshop.testkit.driver.adapter.external.clock.client.dtos.ExtGetTimeResponse;
import com.mycompany.myshop.testkit.driver.adapter.external.clock.client.dtos.error.ExtClockErrorResponse;

import java.time.Instant;

public class ClockRealClient {
    public Result<Void, ExtClockErrorResponse> checkHealth() {
        now();
        return Result.success();
    }

    public Result<ExtGetTimeResponse, ExtClockErrorResponse> getTime() {
        var response = ExtGetTimeResponse.builder()
                .time(now())
                .build();
        return Result.success(response);
    }

    private static Instant now() {
        return Instant.now();
    }
}
